src/replicon_core/replication_rules.rs

```diff
@@ -134,8 +134,8 @@ impl ReplicationRules {
     pub(crate) unsafe fn get_info_unchecked(
         &self,
         replication_id: ReplicationId,
-    ) -> &ReplicationInfo {
-        self.info.get_unchecked(replication_id.0)
+    ) -> ReplicationInfo {
+        *self.info.get_unchecked(replication_id.0)
     }
```

Comment:
In C++ I would return it by reference no question. But how it's in Rust? Docs mentioned that if something can implement `Copy`, then it probably should. But does it mean that it should be still carefully passed by reference when not needed?

Reply:
The following command finishes fine:
```
cargo clippy -- -D clippy::large_types_passed_by_value
```

Reply:
If it's better to pass by reference, then I would remove `Copy`.

Reply:
> Does that clippy pierce unsafe code?

Can't say for sure.

I also found that in Rust before 1.0 `Copy` [was implemented for all types](

Does that clippy pierce unsafe code?
https://github.com/rust-lang/rust/commit/096a28607fb80c91e6e2ca64d9ef44c4e550e96c) that can implement it automatically.

Reply:
From what I found in the internet, if something implements `Copy`, its passed by value.
I would remove `Copy` if we want to pass it by reference to make it explicit to avoid accidental copying.